lib/src/http/common/client.dart

```diff
@@ -93,7 +104,11 @@ abstract class CommonClient implements Client {
     if (withCredentials == true) {
       request.withCredentials = true;
     }
-    _requests.add(request);
+    request.autoRetry
+      ..enabled = autoRetry.enabled
+      ..forHttpMethods = autoRetry.forHttpMethods
+      ..forStatusCodes = autoRetry.forStatusCodes
+      ..test = autoRetry.test;
```

Comment:
I actually don't want to clone the auto retry object between retries - I want the same instance to be shared so that the contextual information (like the current number of attempts) is correct regardless of which request instance you have a reference to.